### ISSUE TITLE
ValidateNested not working

### ISSUE THREAD

**Issue by rugggger** [typestack/class-validator#520]
```
import { IsInt, IsString, IsMongoId, IsBoolean, ValidateNested, IsDefined} from 'class-validator';

export class nestedLesson {
  @IsInt() lesson_number: number;
}

export class Course {
  @IsString()
  @IsMongoId()
  _id: string;
  @IsInt({
    message: "seqNo must be numeric"
  })
  seqNo: number;
  @IsString() description: string;
  @ValidateNested() @IsDefined() lesson: nestedLesson
}
```

In the above example all validation rules work - even the last one @IsDefined for the nestedLesson class.
Only ValidateNested doesn't throw an error when appropriate. 

(for example , for 
{
  "seqNo": 344,
"lesson": {
   "lesson_number": "dfsdfsdf"
}
}

**Comment by juliomsilva** [comment]
try this:

```js
import { Type } from 'class-transformer';

@Type(() => nestedLesson) @ValidateNested() @IsDefined() lesson: nestedLesson
```

**Comment by rugggger** [comment]
Thanks Julio.

It would be nice if this could be added to the official documentation...

**Comment by arczi87** [comment]
It doesn't work form me even with  Type decorator. 
It gives me error during the compilation

`Cannot access 'DailyMealPlan' before initialization`

**Comment by rugggger** [comment]
You need to move the definition of DailyMealPlan up, before you refer to it.
